docs/architecture.md

```diff
@@ -23,67 +23,87 @@
 
 ## Core components
 
-### Connections
+### Protocol
 
-A connection allows the AEA to connect to an external service which has a Python SDK or API. A connection wraps an external SDK or API.
+Protocols define how messages are represented and encoded for transport. They also define the rules to which messages have to adhere in a message sequence. 
 
-### Skill
+For instance, a protocol may contain messages of type `START` and `FINISH`. From there, the rules could prescribe that a message of type `FINISH` must be preceded by a message of type `START`.
 
-A skill can encapsulate any code and ideally delivers economic value to the AEA. Each skill has at most a single Handler and potentially multiple Behaviours and Tasks. The Handler is responsible for dealing with messages of the protocol type for which this skill is registered, as such it encapsulates `reactions`. A Behaviour encapsulates `actions`, that is sequences of interactions with other agents initiated by the AEA. Finally, a Task encapsulates background work which is internal to the AEA.
+The `Message` class in the `protocols/base.py` module provides an abstract class with all the functionality a derived Protocol message class requires for a custom skill, such as basic message generating and management functions and serialisation details.
```

Comment:
Is there a way to link to code with mkdocs?

Reply:
change '...class requires for a custom skill' to '..class requires for a custom protocol'

Reply:
we could add a list of the protocols distributed with the framework:
- oef
- default
- fipa

Reply:
"Is there a way to link to code with mkdocs?" We can embed code snippets, and there is a nice way to embed any tabbed multi-code snippets. However, there is no autogenerate documentation from code function like Sphinx or autodocs.

Reply:
Further to the above, given the code is notated, we can run sphinx on it and somehow embed the auto-generated code docs into the documentation site. We are doing this now with OEF although we need to work on it more.